### ISSUE TITLE
Add missing most common errors 

### ISSUE THREAD

**Issue by OMotornyi** [RobertCraigie/prisma-client-py#349]
## Problem

There are certain [query engine errors](https://www.prisma.io/docs/reference/api-reference/error-reference#prisma-client-query-engine) that are not mapped to the python client errors. A [foreign key violation error](https://www.prisma.io/docs/reference/api-reference/error-reference#p2003) would be notable example of a very common error that might require a dedicated handling in the code. Currently we are getting DataError with the following message:
```python
prisma.errors.DataError: Foreign key constraint failed on the field: `some_fild_fkey` (index)
```

## Suggested solution

Add proper query engine error mapping to https://github.com/RobertCraigie/prisma-client-py/blob/main/src/prisma/engine/utils.py and corresponding test.

## Alternatives

The only alternative would be to parse actual error message `Foreign key constraint failed on the field` which to me feels like a fragile solution.

**Comment by RobertCraigie** [comment]
@jacobdr Yeah I want to keep this open for now. Not mentioned in the issue description is that we need to investigate what errors are actually exposed upstream to see what we should support, once that is done this issue can be closed.
